src/test/java/org/jenkinsci/plugins/durabletask/BourneShellScriptTest.java

```diff
@@ -83,25 +92,132 @@
+            case UBUNTU:
+                container = dockerUbuntu.get();
+                break;
+            default:
+                Assert.fail("Unknown enum value: " + platform);
+                break;
+        }
+        return createDockerSlave(container, customJavaPath);
+    }
+
+    private Slave prepareAgentCommandLauncher() throws Exception{
+        // counter used to prevent name smashing when a docker container from the previous test is
+        // still shutting down but the new test container is being spun up. Seems more ideal than adding a wait.
+        String name = "docker";
+        String agent = "agent-" + counter++;
+        String remoteFs = "/home/jenkins/" + agent;
+
+        Slave agentSlave = null;
+        switch (platform) {
+            case SIMPLE:
+                String dockerRunSimple = String.format("docker run -i --rm --name %s jenkinsci/slave:3.7-1 java -jar /usr/share/jenkins/slave.jar", agent);
+                agentSlave = new DumbSlave(name, remoteFs, new SimpleCommandLauncher(dockerRunSimple));
+                break;
+            case TINI:
+                String dockerRunTini = String.format("docker run -i --rm --name %s --init jenkinsci/slave:3.7-1 java -jar /usr/share/jenkins/slave.jar", agent);
+                agentSlave = new DumbSlave(name, remoteFs, new SimpleCommandLauncher(dockerRunTini));
+                break;
+            default:
+                Assert.fail("Unknown enum value: " + platform);
+                break;
+        }
+        return agentSlave;
+    }
+
+    @After public void slaveCleanup() throws IOException, InterruptedException {
+        s.toComputer().disconnect(new OfflineCause.UserCause(null, null));
+        j.jenkins.removeNode(s);
+    }
+
+    @Test public void smokeTest() throws Exception {
+        boolean notNative = (platform != TestPlatform.NATIVE);
+        int sleepSeconds = 0;
+        if (notNative) {
+            switch (platform) {
+                case CENTOS:
+                case UBUNTU:
+                case TINI:
+                case SIMPLE:
+                    sleepSeconds = 10;
+                    break;
+                case ALPINE:
+                case SLIM:
+                    sleepSeconds = 45;
+                    break;
+                default:
+                    Assert.fail("Unknown enum value: " + platform);
+                    break;
+            }
+        }
+
+        String script = String.format("echo hello world; sleep %s", sleepSeconds);
+        Controller c = new BourneShellScript(script).launch(new EnvVars(), ws, launcher, listener);
         awaitCompletion(c);
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
         c.writeLog(ws,baos);
         assertEquals(0, c.exitStatus(ws, launcher, listener).intValue());
         assertTrue(baos.toString().contains("hello world"));
         c.cleanup(ws);
+        if (notNative) {
+            assertTrue("no zombies running", noZombies());
```

Comment:
So the binaries leave zombie processes behind only in this test? Any idea why? Would be nice to have a comment in the test explaining the reason.

Reply:
It was mostly just mimicking how the tests were running in the previous iteration, but I wonder if  we need to explicitly do this.  Will expend on this thought.

Reply:
My question was more about why this specific test is looking at `nonNative` while the others are all looking at `simpleOrTini()`, since it looks like before this change there was no code to check for zombies at all anywhere, so it's not obvious to me what is special about the native platform for this test vs the other tests.

No strong opinion on whether we want the zombie tests or not. If they give use useful information on the other platforms, then seems fine to leave them in to keep us from accidentally making a change that regresses the behavior. If they are pointless because all of the docker fixtures pass `--init` and so it doesn't matter what our script does, then it seems better to remove them.

Reply:
Yeah, to be honest I'm not sure how my tests evolved that way to not checking for zombies in native form. I may have been emulating how the original tests only did  a zombie check for smoke tests when running on a docker platform. 

My plan is to remove the zombie checks completely, and then see if i can add `UBUNTU_NO_BINARY` platform.

Reply:
> the docker agent now runs with `--init`

I changed the documentation to _recommend_ that it does. Except in Kubernetes packagings, we cannot control for sure how users run their agents. There was still a bug in this plugin, which I fixed and wrote a test for, so I recommend the tests continue to verify that no zombies are left behind.

Reply:
@jglick  thanks for the input, I'll go ahead and keep the zombie tests.  However, I think I can remove the `tini` container since we know it is reaping zombies, we don't need to test for that.  I'll also just go ahead and test for zombies only when running the `simple` container

@dwnusbaum Just fyi, I think I do remember now why I removed the zombie test for native. It was because the way that zombies were being checked works for a relatively clean docker container, but in a process-noisy native environment, the test always seems to catch phantom zombies.

Reply:
Tests have been moved to a separate PR.  please see #103